src/main/java/ch/njol/skript/expressions/ExprLastSpawnedEntity.java

```diff
@@ -102,13 +108,18 @@ protected Entity[] get(Event e) {
 			case 3:
 				en = EffLightning.lastSpawned;
 				break;
+			case 4:
+				en = EffFireworkLaunch.lastSpawned;
+				break;
 			default:
 				en = null;
 		}
+
 		if (en == null)
 			return null;
 		if (!type.isInstance(en))
 			return null;
+
 		Entity[] one = (Entity[]) Array.newInstance(type.getType(), 1);
 		one[0] = en;
 		return one;
```

Comment:
Can't this just use CollectionUtils.array()?

Reply:
I couldn't get that working due to generics
![image](https://user-images.githubusercontent.com/20037329/185722358-1b797a3b-57ef-4a7b-93b2-3eaa7867e485.png)

Reply:
You have to add a cast, exactly like how it was done before.

Reply:
I have tried, it didn't work
![image](https://user-images.githubusercontent.com/20037329/185724054-5f9ff112-684a-47d9-976c-910d10b4e5f9.png)

Reply:
It's `CollectionUtils.array(entity)` not it's type.

Reply:
That will cause this [fail](https://github.com/SkriptLang/Skript/runs/7927683613?check_suite_focus=true)

Reply:
The only difference I see here, is that it'll return it's subtype, rather than just Entity before. We don't use Firework entity type anywhere in Skript so this new array based on it's type isn't really required in my opinion.

Reply:
Returning subtype is important for other expressions/callers to get the actual type of the return rather that its supertype which will cause CCE as seen, if you have a better solution feel free to suggest.